packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,396 @@
+
+      expect(checkbox).not.toHaveAttribute('aria-invalid');
+    });
+
+    it('can have an invalid state', async () => {
+      const { container } = await render('<input [invalid]="invalid" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          invalid: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]');
+
+      expect(checkbox).toBeInvalid();
+    });
+
+    it('can have a invalid state in CSS', async () => {
+      const { container } = await render('<input [required]="required" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          required: true,
+        },
+      });
+
+      const checkbox = container.querySelector(':invalid');
+
+      expect(checkbox).toBeInTheDocument();
+      expect(checkbox).toBeInvalid();
+    });
+  });
+
+  describe('disabled variant', () => {
+    it('renders a design system BEM modifier class name', async () => {
+      const { container } = await render('<input [disabled]="disabled" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          disabled: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]:disabled');
+
+      expect(checkbox).toHaveClass('utrecht-checkbox--disabled');
+    });
+
+    it('is not disabled by default', async () => {
+      await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      const checkbox = screen.getByRole('checkbox');
+
+      expect(checkbox).not.toBeDisabled();
+    });
+
+    it('omits non-essential disabled attributes when not disabled', async () => {
+      await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      const checkbox = screen.getByRole('checkbox');
+
+      expect(checkbox).not.toHaveAttribute('aria-disabled');
+
+      expect(checkbox).not.toHaveAttribute('disabled');
+    });
+
+    it('can have a disabled state', async () => {
+      const { container } = await render('<input [disabled]="disabled" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          disabled: true,
+        },
+      });
+
+      const checkbox = container.querySelector('input[type="checkbox"]:disabled');
+
+      expect(checkbox).toBeDisabled();
+    });
+
+    it('can have a disabled state in CSS', async () => {
+      const { container } = await render('<input [disabled]="disabled" type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+        componentProperties: {
+          disabled: true,
+        },
+      });
+      const checkbox = container.querySelector('input[type="checkbox"]:disabled');
```

Comment:
I think that we can keep this selector simple, like you did in the other tests: `querySelector(':disabled')`